### PR TITLE
New master+wip

### PR DESCRIPTION
**Pull request by acomodi**
Syncs with upstream master and adds the `wip/blackbox_timing` branch.

### CODE REVIEW

**Comment by litghost** [comment]
Why are we adding `wip/blackbox_timing` versus just working to get this on the upstream repo?  We are very close to have no downstream changes, and I don't understand the impulse to create this master+wip with a new `wip/` branch versus just getting this on upstream?

**Comment by acomodi** [comment]
Updated master+wip with the newest master sync.

This includes three wip branches:
- connection box
- issue template
- symbiflow badger

`Update golden results` branch is not needed anymore as the base cost calculation bug is now gone. It was tampering the results of the strong tests, and now with the bug-fix, the QoR are passing locally.

**Review by litghost — APPROVED** [review]
Code looks good, but CI is red?

**Review by litghost — DISMISSED** [review]
CI is red?

**Review by litghost — CHANGES_REQUESTED** [review]
CI is red!?

**Comment by acomodi** [comment]
> CI is red!?

VTR + archdefs is red due to the unroutability [issue](https://github.com/verilog-to-routing/vtr-verilog-to-routing/issues/1571):
```
## Initializing router criticalities took 0.00 seconds (max_rss 3252.2 MiB, delta_rss +0.0 MiB)
---- ------ ------- ---- ------- ------- ------- ----------------- --------------- -------- ---------- ---------- ---------- ---------- --------
Iter   Time    pres  BBs    Heap  Re-Rtd  Re-Rtd Overused RR Nodes      Wirelength      CPD       sTNS       sWNS       hTNS       hWNS Est Succ
      (sec)     fac Updt    push    Nets   Conns                                       (ns)       (ns)       (ns)       (ns)       (ns)     Iter
---- ------ ------- ---- ------- ------- ------- ----------------- --------------- -------- ---------- ---------- ---------- ---------- --------
Warning 104: No routing path for connection to sink_rr 2738315, retrying with full device bounding box
Warning 105: No routing path for connection to sink_rr 2363413, retrying with full device bounding box
Cannot route from BLK-TL-RIOPAD_M[0].IOI_ILOGIC0_O[0] (RR node: 1485596 type: SOURCE location: (114,113) class: 53 capacity: 1 fan-in: 0 fan-out: 1) to BLK-TL-CLBLL_L[0].CLBLL_LL_A2[0] (RR node: 2363413 type: SINK location: (11,143) class: 1 capacity: 1 fan-in: 1 fan-out: 0) -- no possible path
Failed to route connection from 'sw[14]' to '$abc$2260$lut$iopadmap$led[11].T0' for net '$techmap2284$abc$2260$lut$iopadmap$led[11].A[1]' (#1)
Routing failed for net 1
# Routing took 1.89 seconds (max_rss 3252.2 MiB, delta_rss +0.0 MiB)
Circuit is unroutable with a channel width factor of 500.
For a detailed report on the RR node overuse information (report_overused_nodes.rpt), specify --generate_rr_node_overuse_report on.
```

While strong and nightly fail due to a python issue, unsure why this is happening as the upstream master doesn't seem to have this issue:
```
Traceback (most recent call last):
  File "/tmpfs/src/github/vtr-verilog-to-routing/vtr_flow/scripts/run_vtr_flow.py", line 16, in <module>
    import vtr
  File "/tmpfs/src/github/vtr-verilog-to-routing/vtr_flow/scripts/python_libs/vtr/__init__.py", line 4, in <module>
    from .util import (
  File "/tmpfs/src/github/vtr-verilog-to-routing/vtr_flow/scripts/python_libs/vtr/util.py", line 13, in <module>
    from prettytable import PrettyTable
ImportError: No module named 'prettytable'
```

**Comment by litghost** [comment]
> While strong and nightly fail due to a python issue, unsure why this is happening as the upstream master doesn't seem to have this issue:

We should figure out what this is, and fix it.

**Comment by litghost** [comment]
What is the status of this PR?  Should likely re-create now that https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/1579 has been merged?

**Comment by acomodi** [comment]
@litghost Yes, this needs to be updated once again. I will probably need to add a new wip/ branch (or modify an existing one) to solve the python issues, as I think that some python dependencies are not installed correctly.

**Comment by litghost** [comment]
> @litghost Yes, this needs to be updated once again. I will probably need to add a new wip/ branch (or modify an existing one) to solve the python issues, as I think that some python dependencies are not installed correctly.

Why do we need a wip/ branch for?  If a python dependency is missing, just add it to the upstream requirements.txt?

**Comment by acomodi** [comment]
Let me rephrase, the python dependencies is already in the requirements, the problem is that it seems that it does not get installed correctly. Upstream Kokoro builds are working fine. Maybe it was a one-time issue though, I'll need to re-run this with the updated master+wip

**Comment by acomodi** [comment]
I think I know what might be happening here with the failing Nightly and Strong Kokoro CIs.

I didn't realize that Kokoro in this repository uses https://github.com/SymbiFlow/vtr-verilog-to-routing-kokoro, and possibly the issue needs to be solved in there

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/SymbiFlow/vtr-verilog-to-routing-kokoro">SymbiFlow/vtr-verilog-to-routing-kokoro</a></strong></div><div>Kokoro Configuration to run against SymbiFlow/vtr-verilog-to-routing repository. - SymbiFlow/vtr-verilog-to-routing-kokoro</div></blockquote>

**Comment by acomodi** [comment]
Strong tests in Kokoro turned out green finally

**Comment by acomodi** [comment]
```
## Initializing router criticalities took 0.00 seconds (max_rss 3251.9 MiB, delta_rss +0.0 MiB)
---- ------ ------- ---- ------- ------- ------- ----------------- --------------- -------- ---------- ---------- ---------- ---------- --------
Iter   Time    pres  BBs    Heap  Re-Rtd  Re-Rtd Overused RR Nodes      Wirelength      CPD       sTNS       sWNS       hTNS       hWNS Est Succ
      (sec)     fac Updt    push    Nets   Conns                                       (ns)       (ns)       (ns)       (ns)       (ns)     Iter
---- ------ ------- ---- ------- ------- ------- ----------------- --------------- -------- ---------- ---------- ---------- ---------- --------
Warning 104: No routing path for connection to sink_rr 2320172, retrying with full device bounding box
Warning 105: No routing path for connection to sink_rr 2853099, retrying with full device bounding box
Cannot route from BLK-TL-RIOPAD_M[0].IOI_ILOGIC0_O[0] (RR node: 1485596 type: SOURCE location: (114,113) class: 53 capacity: 1 fan-in: 0 fan-out: 1) to BLK-TL-CLBLM_R[0].CLBLM_L_A2[0] (RR node: 2853099 type: SINK location: (105,139) class: 51 capacity: 1 fan-in: 1 fan-out: 0) -- no possible path
Failed to route connection from 'sw[14]' to '$abc$2260$lut$iopadmap$led[11].T0' for net '$techmap2284$abc$2260$lut$iopadmap$led[11].A[1]' (#1)
Routing failed for net 1
# Routing took 1.86 seconds (max_rss 3251.9 MiB, delta_rss +0.0 MiB)
Circuit is unroutable with a channel width factor of 500.
For a detailed report on the RR node overuse information (report_overused_nodes.rpt), specify --generate_rr_node_overuse_report on.
```

arch-defs CI will be fixed once https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1711 will be merged

**Comment by litghost** [comment]
Nightly is red?

**Comment by acomodi** [comment]
> Nightly is red?

That failure is due to https://github.com/SymbiFlow/vtr-verilog-to-
routing/pull/549/commits/bfa6fdb0e626c9e55f35853d7db7b94a1cfd59f8.

We have been merging previous master+wip with Nightly failing as well, I might as well update the golden results for that one also so it turns out green, at least until we have the base cost fix in place.


<blockquote><div> GitHub</div><div><strong><a href="https://github.com">Build software better, together</a></strong></div><div>GitHub is where people build software. More than 50 million people use GitHub to discover, fork, and contribute to over 100 million projects.</div></blockquote>
